genai-perf/tests/test_data_parser/test_llm_profile_data_parser.py

```diff
@@ -1355,3 +1355,432 @@ def test_no_session_metrics(self, mock_json) -> None:
+            (
+                "openai",
+                ResponseFormat.HUGGINGFACE_GENERATE,
+                '{"generated_text":"Hello, world!"}',
+                None,  # Will raise ValueError
+            ),
+            # OpenAI chat completion
+            (
+                "openai",
+                ResponseFormat.OPENAI_CHAT_COMPLETIONS,
+                '{"object":"chat.completion","choices":[{"message":{"content":"Hello, world!"}}]}',
+                "Hello, world!",
+            ),
+            # OpenAI chat completion streaming
+            (
+                "openai",
+                ResponseFormat.OPENAI_CHAT_COMPLETIONS,
+                '{"object":"chat.completion.chunk","choices":[{"delta":{"content":"Hello, world!"}}]}',
+                "Hello, world!",
+            ),
+            # OpenAI chat completion without object field (vLLM workaround)
+            (
+                "openai",
+                ResponseFormat.OPENAI_CHAT_COMPLETIONS,
+                '{"choices":[{"text":"Hello, world!"}]}',
+                "Hello, world!",
+            ),
+            # OpenAI completion
+            (
+                "openai",
+                ResponseFormat.OPENAI_COMPLETIONS,
+                '{"object":"text_completion","choices":[{"text":"Hello, world!"}]}',
+                "Hello, world!",
+            ),
+            # OpenAI completion without object field (vLLM workaround)
+            (
+                "openai",
+                ResponseFormat.OPENAI_COMPLETIONS,
+                '{"choices":[{"text":"Hello, world!"}]}',
+                "Hello, world!",
+            ),
+            # Triton generate
+            (
+                "triton",
+                ResponseFormat.TRITON_GENERATE,
+                '{"text_output":"Hello, world!"}',
+                "Hello, world!",
+            ),
+            # Empty response
+            (
+                "openai",
+                ResponseFormat.OPENAI_CHAT_COMPLETIONS,
+                "",
+                "",
+            ),
+            # [DONE] response
+            (
+                "openai",
+                ResponseFormat.OPENAI_CHAT_COMPLETIONS,
+                "data: [DONE]",
+                "",
+            ),
+            # Non-data SSE field
+            (
+                "openai",
+                ResponseFormat.OPENAI_CHAT_COMPLETIONS,
+                ": ping",
+                "",
+            ),
+            # Unknown format
+            (
+                "openai",
+                "UNKNOWN_FORMAT",  # type: ignore
+                "any response",
+                None,  # Will raise ValueError
+            ),
+        ],
+    )
+    @patch("genai_perf.profile_data_parser.profile_data_parser.load_json")
+    def test_extract_text_output(
+        self, mock_json, service_kind, response_format, response, expected_text
+    ) -> None:
+        """Test the text extraction methods for different response formats."""
+        if service_kind == "triton":
```

Comment:
Actually on a second thought, can't we simplify the test case by getting rid of this entirely? Since our goal for this test is to verify that this single `_extract_text_output` method is working properly, we can just instantiate `LLMProfileDataParser` with any pre-defined json variable (e.g. `openai_profile_data`), manually change response format to our target format (which the test already does), and call `_extract_text_output` with target response string.

This makes sense because testing `_extract_text_output` doesn't (and shouldn't) require the knowledge of profile export json format and therefore the test doesn't need to redefine the return values of mock_json for all the different service_kinds.

Reply:
Good thinking, done! The test is more straightforward now.